### DISCUSSION TITLE
Release v0.23.0-RC  🏗️ Almost there 🏗️

### DISCUSSION THREAD

**Discussion by simulot** [simulot/immich-go#539] [Announcements]
# Release v0.23.0-RC  🏗️ Almost there 🏗️

This version is a complete rewrite of the original immich-go project. It is designed to be more efficient, more reliable, and easier to use. It is also more flexible, with more options and features.

What's new:
- archive
- tags

I'd love to hear your feedback!

 Share your thoughts: 
- Let me know what you think in  the discussion, or on Discord.
- If you're up for it, please take a look at the README file and let me know if you spot any typos or unclear sections.

<hr /><em>This discussion was created from the release <a href='https://github.com/simulot/immich-go/releases/tag/v0.23.0-RC'>Release v0.23.0-RC  🏗️ Almost there 🏗️</a>.</em>

**Comment by latetedemelon** [comment]
Hi @simulot,

Thank you for creating and maintaining such a fantastic project! It's been a great help as I consolidate my multiple Google accounts. While testing the new version, I encountered a few quirks that might be worth looking into (or perhaps they're intended behavior).

Observations:

1. XMP file issue - When uploading a duplicate image, it seems to throw an error if an XMP file isn't already present. I would have assumed it should create one automatically, but this might be the intended behavior.

1. Resume functionality - I've noticed that canceling and resuming uploads seems to restart the process from the beginning rather than picking up where it left off. I believe the feature is supposed to resume, so I wanted to flag this.

1. Duplicate removal process - Regarding the removal of duplicates, could you confirm if the process consolidates albums as follows? My assumption is that it:

- Reviews all duplicates of an image.
- Adds the highest-resolution version to all albums containing duplicates.
- Checks for a sidecar on the highest-resolution image.
- Populates or creates the sidecar with data from other duplicates.
- Deletes the duplicates and associated sidecars.

If any of these points are already addressed or part of your roadmap, no worries—I just wanted to share what I observed. Thank you again for all the hard work on this project!

```
immich_server            | [Nest] 7  - 11/26/2024, 10:59:20 PM    WARN [Microservices:Error: File not found - /usr/src/app/upload/upload/1726445a-de4b-48b8-8967-5fed9e23a7c8/6b/2b/6b2b6e82-1e89-404f-a6f9-e733e78f1c55.MOV.xmp
immich_server            |     at ReadTask._ExifToolTask_parser (/usr/src/app/node_modules/exiftool-vendored/dist/ExifToolTask.js:76:71)
immich_server            |     at ReadTask.parser (/usr/src/app/node_modules/exiftool-vendored/dist/ExifToolTask.js:47:167)
immich_server            |     at ReadTask._Task_resolve (/usr/src/app/node_modules/batch-cluster/dist/Task.js:146:40)
immich_server            |     at runNextTicks (node:internal/process/task_queues:65:5)
immich_server            |     at process.processTimers (node:internal/timers:526:9)] Error reading exif data (upload/upload/1726445a-de4b-48b8-8967-5fed9e23a7c8/6b/2b/6b2b6e82-1e89-404f-a6f9-e733e78f1c55.MOV.xmp): Error: File not found - /usr/src/app/upload/upload/1726445a-de4b-48b8-8967-5fed9e23a7c8/6b/2b/6b2b6e82-1e89-404f-a6f9-e733e78f1c55.MOV.xmp
```

**Comment by latetedemelon** [comment]
Sorry, one further clarification that's related.  When uploading a duplicate photo, does the project update immich with any previously missing album and metadata (e.g. GPS) that was previously not added?

Thanks.

**Comment by simulot** [comment]
Thank for testing the project!

**XMP**: Probably an untested situation. Could you describe how to reproduce the problem?

**Resume**: Immich-go don't retain anything from the previous runs. It's intentional, and nothing prevent the user to change something in the input. So, immich-go restart from the beginning, and skip photos already present on the server.  

**Duplicates and XMP**
Probably not anticipated.

Immich-go considers duplicates files with the same name, same bytes length. When a file comes with same name, same length, it is skipped. 

When immich own a photo with the same name, and with the same date of capture, their length are compared. The server is updated to keep only the heaviest file, supposed to be the high resolution version.
 
For google photos takeouts only, a full scan of the takeout is performed before starting the upload. The reference of the duplicated photo is added to each album.
Folder import is managed folder by folder.

Metadata: 
Immich-go doesn't use metadata at the moment other than the date of capture. 
Maybe a future feature.

**Reply by latetedemelon** [reply]
It may be my misunderstanding, but I understood that Immich-go imported the Google sidecar aka metadata as part of its core process?  Immich proper stores this in an XMP file I believe.

The issue appears to happen when the (duplicate) file exists on the Immich server without an xmp file.  Immich-go appears (from what I can tell) to attempt to write to the file and fails when processing a duplicate.  Ideally obviously I'd like it to update the file with the photo date and gps at a minimum.

**Comment by simulot** [comment]
Bonjour, let me know if we can continue in French as your nickname suggests.

I can't reproduce your case:

Run 1: Upload a file without a XMP file 
Run 2: Upload the same file with an XMP file.

The 2nd run doesn't raise any error, other than signaling the duplicate.

**Comment by latetedemelon** [comment]
Sorry, I should clarify it’s Immich throwing the errors.  I’ll go back and
attempt to nail down reproducing the error.

I’d love to be able to speak French but I’ve just got my grade school in
education to fall back in there (Canadian).  My username is an inside joke
amongst my friends.
